### PR TITLE
Fix commtrack report

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?157564#876066

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/reports/commtrack/data_sources.py [hunk 1]
```diff
@@ -434,19 +434,18 @@ def get_data(self):
 
                 results = StockReport.objects.filter(
                     stocktransaction__case_id=spoint_id
+                ).filter(
+                    date__gte=self.converted_start_datetime,
+                    date__lte=self.converted_end_datetime
                 ).values_list(
                     'form_id',
                     'date'
-                ).order_by('-date').distinct()  # not truly distinct due to ordering
```
Comment (emord) [inline corehq/apps/reports/commtrack/data_sources.py]: Looks like order was only needed for the not matched case, which uses a new query

File: corehq/apps/reports/commtrack/data_sources.py [hunk 2]
```diff
@@ -434,19 +434,18 @@ def get_data(self):
...
+                ).distinct()  # not truly distinct due to ordering
 
                 matched = False
                 for form_id, date in results:
-                    if self.converted_start_datetime > date:
```
Comment (emord) [inline corehq/apps/reports/commtrack/data_sources.py]: No need to check dates if we move it to the sql query

**Comment by czue** [comment]
looks good to me though will let @twymer review/merge since it's his code

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Pull request merged**